contract/r/gnoswap/launchpad/state.gno

```diff
@@ -0,0 +1,95 @@
+package launchpad
+
+import (
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	// projectId -> Project
+	projects *avl.Tree
+
+	// projectTierId -> RewardManager
+	projectTierRewardManagers *avl.Tree
+
+	// Counter for generating unique deposit IDs
+	depositCounter *Counter
+
+	// depositId -> Deposit, Tree storing all deposits by ID
+	deposits *avl.Tree
+)
+
+func init() {
+	projects = avl.NewTree()
+	projectTierRewardManagers = avl.NewTree()
+
+	depositCounter = NewCounter()
+	deposits = avl.NewTree()
+}
+
+func getProject(projectID string) (*Project, error) {
+	project, ok := projects.Get(projectID)
+	if !ok {
+		return nil, makeErrorWithDetails(errDataNotFound, ufmt.Sprintf("project(%s) not found", projectID))
+	}
+
+	return project.(*Project), nil
```

Comment:
I think we need to always check the boolean to verify that the conversion was done correctly, especially converting an interfaces. for example:
```go
	p, ok := project.(*Project)
	if !ok {
		// returning error here
	}
	return p, nil
```

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/777/commits/759ea8dc2cebbf08bacd5076f2578e3bebb2d37f